src/parser.rs

```diff
@@ -1,31 +1,41 @@
-use crate::ParseError;
+use crate::{propagate, ParseError};
 
-pub fn take<const N: usize>(input: &[u8]) -> Result<(&[u8], [u8; N]), ParseError> {
-    if let (Some(value), Some(rest)) = (input.get(0..N), input.get(N..)) {
-        Ok((rest, value.try_into().unwrap()))
+pub const fn take2(input: &[u8]) -> Result<(&[u8], [u8; 2]), ParseError> {
```

Comment:
This module could be simplified a bit when we decide to raise the MSRV. We usually leave a comment with a `MSRV: ` prefix in the code as a reminder to revisit the code:
```rust
// MSRV: Use `split_at_checked` (1.80.0) and `split_first_chunk` (1.77.0) in this module.
```

Reply:
Yeah I thought about using `split_first_chunk`, but 1.77 seems very recent.